src/mlpack/methods/random_forest/bootstrap.hpp

```diff
@@ -43,6 +45,153 @@ void Bootstrap(const MatType& dataset,
     bootstrapWeights = weights.cols(indices);
 }
 
+/**
+ * Default bootstrap strategy that uses the ::Bootstrap() function. 
+ */
+template<bool UseWeights>
+class DefaultBootstrap
+{
+ public:
+  template<typename MatType,
+           typename LabelsType,
+           typename WeightsType>
+  void Bootstrap(const MatType& dataset,
+                 const LabelsType& labels,
+                 const WeightsType& weights,
+                 MatType& bootstrapDataset,
+                 LabelsType& bootstrapLabels,
+                 WeightsType& bootstrapWeights)
+  {
+    mlpack::Bootstrap<UseWeights>(dataset,
+                                  labels,
+                                  weights,
+                                  bootstrapDataset,
+                                  bootstrapLabels,
+                                  bootstrapWeights);
+  }
+};
+
+/**
+ * No-op bootstrap strategy that just copies the input to the output.
+ */
+template<bool UseWeights>
+class IdentityBootstrap
+{
+ public:
+  template<typename MatType,
+           typename LabelsType,
+           typename WeightsType>
+  void Bootstrap(const MatType& dataset,
+                 const LabelsType& labels,
+                 const WeightsType& weights,
+                 MatType& bootstrapDataset,
+                 LabelsType& bootstrapLabels,
+                 WeightsType& bootstrapWeights)
+  {
+    bootstrapDataset = dataset;
+    bootstrapLabels = labels;
+    if (UseWeights)
+      bootstrapWeights = weights;
```

Comment:
These will incur copies of the dataset, but this is actually not necessary; we can make an [alias](https://www.mlpack.org/doc/user/core/math.html#aliases):

```suggestion
    MakeAlias(bootstrapDataset, dataset, dataset.n_rows, dataset.n_cols);
    MakeAlias(bootstrapLabels, labels, labels.n_elem);
    if (UseWeights)
      MakeAlias(bootstrapWeights, weights, weights.n_elem);
```